commons/src/main/java/com/github/maleksandrowicz93/educational/institution/common/BaseAggregate.java

```diff
@@ -0,0 +1,15 @@
+package com.github.maleksandrowicz93.educational.institution.common;
+
+import lombok.Getter;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.SuperBuilder;
+
+import static lombok.AccessLevel.PROTECTED;
+
+@SuperBuilder
+@FieldDefaults(makeFinal = true)
+public abstract class BaseAggregate<S extends Snapshot<ID>, ID> implements Aggregate<S, ID> {
+
+    @Getter(PROTECTED)
+    S source;
```

Comment:
Hmmm tym bardzo narzucasz sobie sposób implementacji danego agregatu i że musi być ten snapshot readonly do zapisu.
A może warto też rozważyć zwracanie zdarzeń domenowych jako rezultat i wtedy np. to zapisujesz :) Ja tak robię:
https://event-driven.io/pl/how_events_can_help_on_making_state_based_approach_efficient/

Ewentualnie wzorzec Decider. Ale wiadomo, nie wszystko na raz.